### ISSUE TITLE
Keyboard navigation on Datepicker does not work when is located on a Modal

### ISSUE THREAD

**Issue by chubbywhale** [carbon-design-system/carbon#655]
[Title]: Keyboard navigation on Datepicker does not work when is located on a Modal

## Detailed description

Keyboard navigation (using arrow keys) for selecting date on the calender popup does not work when the Datapicker component is placed within a Modal.  The calendar popup doesn't seem to have focus when it is in view.

Expected the keyboard navigation behaviour to be the same as when the Datepicker component is place on a page rather than in a Modal.

Link to a sample of the issue:
https://codepen.io/anon/pen/RQdmMa/

![image](https://user-images.githubusercontent.com/37340587/37350489-499d9780-26af-11e8-8227-31209f2b3497.png)
